python/ray/data/datasource/parquet_datasource.py

```diff
@@ -56,6 +56,61 @@ def _deregister_parquet_file_fragment_serialization():
     ray.util.deregister_serializer(ParquetFileFragment)
 
 
+# this retry helps when HA hdfs service not able to handle overloaded read request.
```

Comment:
Here and below, can you phrase these in more general way? Think this can be used by non-HDFS users.

Reply:
sure.